app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -35,14 +36,19 @@ def context_urls
 
       def smart_proxy
         @smart_proxy ||= begin
-          proxy = params[:url] ? SmartProxy.unscoped.find_by(url: params[:url]) : SmartProxy.pulp_primary
+          proxy = params[:url] ? find_smart_proxy : SmartProxy.pulp_primary
```

Comment:
Reviewers note: I was a bit tempted to suggest always authenticating the Smart Proxy and seeing if it can, but that can raise warnings. `params[:url]` should be set if a Smart Proxy sends a request so that's good enough for now. In the future we may want to revisit this.